modules/apps/site/site-insurance-site-initializer/remote-web-components/raylife-d2c-web/src/components/containers/Forms/Basics/BusinessInformation/index.js

```diff
@@ -14,41 +12,72 @@ import {PhoneControlledInput} from '../../../../connectors/Controlled/Input/With
 import {ControlledInput} from '../../../../connectors/Controlled/Input';
 import {useCustomEvent} from '../../../../../hooks/useCustomEvent';
 import {TIP_EVENT} from '../../../../../events';
+import {WarningBadge} from '../../../../fragments/Badges/Warning';
 
 const setFormPath = (value) => `basics.businessInformation.${value}`;
 
 export const FormBasicBusinessInformation = () => {
 	const form = useWatch();
+	const [applicationId, setApplicationId] = useState('');
+	const [error, setError] = useState('');
 	const {selectedStep, setSection} = useStepWizard();
 	const [dispatchEvent] = useCustomEvent(TIP_EVENT);
 	const {
 		control,
 		formState: {isValid},
+		setValue,
 	} = useFormContext();
 
 	useEffect(() => {
 		dispatchEvent({
-			templateName: 'hi-template',
 			step: selectedStep,
 			templateData: {
 				firstName: '! 👋',
 			},
+			templateName: 'hi-template',
 		});
 	}, []);
 
+	/**
+	 * @description When the application is created, we set the value to Form Context
+	 * We tried to use setValue directly on goToPrevious and goToNextForm
+	 * and for reasons unknowns, the section is not called.
+	 */
+
+	useEffect(() => {
+		if (applicationId) {
+			setValue('basics.applicationId', applicationId);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [applicationId]);
+
 	const onSave = async () => {
 		try {
-			await LiferayService.createBasicsApplication(form.basics);
+			const response = await LiferayService.createOrUpdateBasicsApplication(
+				form.basics
+			);
+
+			setApplicationId(response.data.id);
 		} catch (error) {
-			console.error(error);
+			setError('Unable to save your information. Please try again.');
+			throw error;
 		}
 	};
 
-	const goToPreviousForm = () =>
+	const goToPreviousForm = async () => {
+		try {
+			if (form.basics?.businessInformation?.business?.email) {
+				await onSave();
+			}
+		} catch (error) {
```

Comment:
Thank you for the feedback, @giu7d!

I have some points to mention: 

* About the try catch on goToPreviousForm, I used it to call `setSection` even if the request fails for saving
* About the throw, I could remove the throw and even the entire try catch block, but I would to use `setError('Unable to save your information. Please try again.');` twice, on: **goToPreviousForm**, **goToNextForm**

Reply:
For option 1: I even thought about using .finally(), because regardless of the result of these requests, I'll want to go to the previous screen, but I don't see it finally() used that often.

For option 2: I don't know what would be better

Do you want to look at those options together?

Reply:
>  I could remove the throw and even the entire try catch block

I mean, remove the `try catch` block from the `goToPreviousForm`. But sure, you need to keep it in the `onSave` in order to treat the error.

>  I used it to call setSection even if the request fails for saving

Ok, but I still have some questions, If we change the function:
```js
const goToPreviousForm = async () => {
	try {
	    if (form.basics?.businessInformation?.business?.email) {
		await onSave();
	   }
	} catch (error) {
	   console.warn(error);
	}
	
	setSection(AVAILABLE_STEPS.BASICS_BUSINESS_TYPE);
};
```

To:

```js
const goToPreviousForm = async () => {
	   if (form.basics?.businessInformation?.business?.email) {
		await onSave();
	   }
	
	setSection(AVAILABLE_STEPS.BASICS_BUSINESS_TYPE);
};
```

Wouldn't it have the same effect?

Reply:
In my perspective, the only reason for this `try catch` to exist is the presence of a `throw error` in line 63 of the `onSave` function